### PR TITLE
EASY-2784: dataverse properties

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-2784: dataverse properties

When applied it will
--------------------
* 
* 
* 

Where should the reviewer @DANS-KNAW/easy start?
------------------------------------------------
With the `*Factory*` classes

How should this be manually tested?
-----------------------------------

Related pull requests on github
-------------------------------

repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala [hunk 1]
```diff
@@ -56,13 +60,18 @@ case class DepositPropertiesFactory(configuration: Configuration) {
...
+        addProperty("dataverse.other-id", "https://doi.org/"+doi)
+      addProperty("dataverse.id-protocol", idType.toString.toLowerCase)
       idType match {
-        case DOI => addProperty("dataverse.identifier", doi)
-        case URN => addProperty("dataverse.identifier", urn)
+        case DOI =>
+          addProperty("dataverse.identifier", doi.replaceAll(".*/", ""))
+          addProperty("dataverse.id-authority", "10.80270")
+        case URN =>
+          addProperty("dataverse.identifier", urn)
```
Comment (lindareijnhoudt) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala:73]:
```suggestion
          addProperty("dataverse.id-identifier", urn)
```
This way the three properties start with the same string

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala [hunk 2]
```diff
@@ -56,13 +60,18 @@ case class DepositPropertiesFactory(configuration: Configuration) {
       addProperty("identifier.urn", urn)
       addProperty("identifier.fedora", fedoraId)
       addProperty("dataverse.bag-id", "urn:uuid:" + bagInfo.uuid)
-      addProperty("dataverse.sword-token", "urn:uuid:" + bagInfo.versionOf.getOrElse(bagInfo.uuid))
-      addProperty("dataverse.nbn", "urn:uuid:" + bagInfo.versionOf.map(getBaseUrn).getOrElse(urn))
+      addProperty("dataverse.sword-token", bagInfo.versionOf.getOrElse(bagInfo.uuid))
+      addProperty("dataverse.nbn", bagInfo.versionOf.map(getBaseUrn).getOrElse(urn))
       if (!configuration.dansDoiPrefixes.contains(doi.replaceAll("/.*", "/")))
-        addProperty("dataverse.other-id", doi)
+        addProperty("dataverse.other-id", "https://doi.org/"+doi)
```
Comment (lindareijnhoudt) [inline src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala:66]: this assumes the `other-id` is a doi. maybe we need to be more careful with this in the future

File: src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala [hunk 3]
```diff
@@ -32,9 +32,8 @@ class AppSpec extends AnyFlatSpec with Matchers with AppConfigSupport with FileS
     }
     new EasyConvertBagToDespositApp(appConfig).addPropsToBags(
       (testDir / "exports").children,
-      IdType.DOI,
       None,
-      DepositPropertiesFactory(appConfig)
+      DepositPropertiesFactory(appConfig, IdType.DOI, BagSource.VAULT)
```
Comment (lindareijnhoudt) [inline src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala:36]: After we remove the SIP-part from `easy-fedora-to-bag` we need some tests here for BagSource.FEDORA as well.

File: src/main/assembly/dist/cfg/application.properties [hunk 4]
```diff
@@ -10,3 +10,4 @@
 
 dans-doi.prefixes=10.17026/,10.5072/
 bag-index.url=http://localhost:20120/
+dataverse.id-authority=10.80270
```
Comment (lindareijnhoudt) [inline src/main/assembly/dist/cfg/application.properties:13]: the value in dtap has a slash at the end. is that problematic here?

File: src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala (hunk 3 above)
Comment (jo-pol) [reply src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala:36]: The option results in not-implemented anyway so far, or don't we need further chages here?

File: src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala (hunk 2 above)
Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.bag2deposit/DepositPropertiesFactory.scala:66]: The `doi` in this context is actually the first `<dcterms:identifier xsi:type="id-type:DOI">`

File: src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala (hunk 3 above)
Comment (lindareijnhoudt) [reply src/test/scala/nl.knaw.dans.easy.bag2deposit/AppSpec.scala:36]: if it comes from FEDORA 
* it shouldn't consult the bag-index,
* the sword token should be empty, or non-existent
* the bag-id is empty
* TYPE should be DOI
* anything else?

**Pull request merged**